WordPress/Classes/Extensions/String+RegEx.swift

```diff
@@ -0,0 +1,62 @@
+import Foundation
+
+
+// MARK: - String: RegularExpression Helpers
+//
+extension String {
+
+    func matches(regex: String, options: NSRegularExpression.Options = []) -> [NSTextCheckingResult] {
+        let regex = try! NSRegularExpression(pattern: regex, options: options)
```

Comment:
Just talked with @jleandroperez about this.

The reason why we want things to stop if the regex is bad, is that we're relying on the regex not being broken for things to work nicely.

We'd definitely prefer a build-time verification of the regexes but that seems not to be supported in Swift.  This is basically the closer we can get to it (coupled with some unit tests we'll be writing for this PR).

The problem is that there's a greater data-corruption risk in trying to take a default-and-silent action than in letting the user and the developers know something's broken.